model/task/task.go

```diff
@@ -993,3 +999,46 @@ func (t *Task) MergeNewTestResults() error {
 	}
 	return nil
 }
+
+// Like Task DependenciesMet, but uses the aggregated results instead of
+// querying the database 1-by-1 for the Tasks
+func (t *DependencyNode) DependenciesMet() bool {
+	if len(t.DependsOn) != len(t.Predecessors) {
+		return false
+	}
+
+	for _, depTask := range t.Predecessors {
+		if !t.satisfiesDependency(&depTask) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func UndispatchedWithEmbeddedDependencies() ([]DependencyNode, error) {
+	pipeline := []bson.M{
+		bson.M{
+			"$match": bson.M{
+				ActivatedKey: true,
+				StatusKey:    evergreen.TaskUndispatched,
+				//Filter out blacklisted tasks
+				PriorityKey: bson.M{"$gte": 0},
+			},
+		},
+		bson.M{
+			"$graphLookup": bson.M{
+				"from":             Collection,
+				"startWith":        "$depends_on._id",
+				"connectFromField": "depends_on._id",
```

Comment:
we have typically build field names in this case using the key variables declared in the package. 

this gets ugly, but tends to be smart.